internal/host/validator.go

```diff
@@ -620,20 +620,35 @@ func (v *validator) isIgnitionDownloadable(c *validationContext) (ValidationStat
 	if c.infraEnv != nil {
 		return ValidationSuccessSuppressOutput, ""
 	}
+
 	if !hostutil.IsDay2Host(c.host) {
 		return ValidationSuccessSuppressOutput, ""
 	}
+
 	if c.host.APIVipConnectivity == "" {
 		return ValidationPending, "Ignition is not yet available, pending API connectivity"
 	}
-	var response models.APIVipConnectivityResponse
-	if err := json.Unmarshal([]byte(c.host.APIVipConnectivity), &response); err != nil || !response.IsSuccess {
-		return ValidationFailure, "Ignition is not downloadable. Please ensure host connectivity to the cluster's API"
+
+	var apiConnectivityResponse models.APIVipConnectivityResponse
+	if err := json.Unmarshal([]byte(c.host.APIVipConnectivity), &apiConnectivityResponse); err != nil {
+		return ValidationError, "Internal error - failed to parse agent API connectivity response"
 	}
-	if swag.BoolValue(c.cluster.UserManagedNetworking) {
-		return ValidationSuccess, "No API connectivity needed: User Managed Networking"
+
+	if apiConnectivityResponse.IsSuccess {
+		return ValidationSuccess, "Ignition is downloadable"
+	}
+
+	if apiConnectivityResponse.URL == "" {
+		// Missing URL means this is a response from an older agent version,
+		// without much information about what went wrong with the download -
+		// so return an undetailed error message
+		return ValidationFailure, "This host has failed to contact the cluster's API, please ensure the host can reach the API"
```

Comment:
We can remove this, now that we have agent upgrade, right?

Reply:
I didn't write the agent part yet, so no

Reply:
I missed that this was new code, my bad